src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -246,7 +246,9 @@ trait LowPriorityConfigConvertImplicits {
 
   implicit val readString = fromString[String](identity)
   implicit val readBoolean = fromString[Boolean](_.toBoolean)
-  implicit val readDouble = fromString[Double](_.toDouble)
+  implicit val readDouble = fromString[Double]({
+    v => if (v.last == '%') v.dropRight(1).toDouble / 100.0 else v.toDouble
```

Comment:
I like the idea of having a `fromNonEmptyString`. I'd suggest defining it more like:

```
def fromNonEmptyString[T](typeName: String)(fromF: String => T): ConfigConvert[T] = new ConfigConvert[T] {
    override def from(config: ConfigValue): Try[T] = fromFConvert {
      case "" => throw new Exception(s"Cannot read a $typeName from an empty string")
      case x => fromF(x)
    }(config)
    override def to(t: T): ConfigValue = ConfigValueFactory.fromAnyRef(t)
}
```

I'd rather see a separate argument list for the function. That'd facilitate writing 
```
fromNonEmptyString("Double") { 
  case v if ... => v.dropRight(1).toDouble
  case v => v.toDouble
}
```

That `_ match {` bit is generally considered bad style, as far as I know. You can just skip straight to the `case`s.